### PR TITLE
Fix few typos and API links

### PR DESCRIPTION
**Pull request by solar05**
Noted that API links in guides goes nowhere, also fix few typos and unclosed bracket.

### CODE REVIEW

File: lib/elixir/pages/references/naming-conventions.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Function names may also start with an underscore. Such functions are never impor
     iex> _wont_be_imported()
     ** (CompileError) iex:1: undefined function _wont_be_imported/0
 
-Due to this property, Elixir relies on functions starting with underscore to attach compile-time metadata to modules. Such functions are most often in the `__foo__` format. For example, every module in Elixir has an [`__info__/1`](`c:Module.__info__/1`) function:
+Due to this property, Elixir relies on functions starting with underscore to attach compile-time metadata to modules. Such functions are most often in the `__foo__` format. For example, every module in Elixir has an [`__info__/1`](https://hexdocs.pm/elixir/main/Module.html#c:__info__/1) function:
```
Comment (wojtekmach) [inline lib/elixir/pages/references/naming-conventions.md:41]: do you have a live example where this link was broken? It works here:

image

(https://hexdocs.pm/elixir/main/naming-conventions.html)

There's quite a bit of logic in ExDoc to make these links work but there're sometimes bugs and we'd love to learn about them so we can fix them.

Comment (wojtekmach) [reply lib/elixir/pages/references/naming-conventions.md:41]: Oh, is it that you're looking at

https://github.com/elixir-lang/elixir/blob/main/lib/elixir/pages/references/naming-conventions.md

and _not_:

https://hexdocs.pm/elixir/main/naming-conventions.html

then yeah, that's a problem.

Comment (wojtekmach) [reply lib/elixir/pages/references/naming-conventions.md:41]: If we'd move forward with this, we should drop main from the url, i.e.:

```diff
- https://hexdocs.pm/elixir/main/naming-conventions.html
+ https://hexdocs.pm/elixir/naming-conventions.html
```

**Comment by josevalim** [comment]
The links are intentional. The main goal is to make them work from ExDoc and not from GitHub. :) However, if you want to submit a PR with the fixed typos and brackets, that would be very welcome!

**Pull request closed without merging**